src/mission_execution/src/mission_executor.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <exception>
+#include "mission_execution/mission_executor.hpp"
+
+using namespace std::chrono_literals;
+using nav2_tasks::TaskStatus;
+
+namespace mission_execution
+{
+
+MissionExecutor::MissionExecutor(const std::string & name)
+: nav2_tasks::ExecuteMissionTaskServer(name)
+{
+  RCLCPP_INFO(get_logger(), "MissionExecutor::MissionExecutor");
+  navigationTask_ = std::make_unique<nav2_tasks::NavigateToPoseTaskClient>("SimpleNavigator", this);

```

Comment:
Minor nit - calling this client "SimpleNavigator" is confusing since the SimpleNavigator class is a server. I guess this is the client side to that, but it could be the client side to any Navigation Server right? Or do I not understand correctly? If it is a client and the server can be provided by something other than the SimpleNavigator server, then can it be "NavigationClient"